openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+    info "Creating slapd.ldif"
+    cat > "${LDAP_SHARE_DIR}/slapd.ldif" << EOF
+#
+# See slapd-config(5) for details on configuration options.
+# This file should NOT be world readable.
+#
+
+dn: cn=config
+objectClass: olcGlobal
+cn: config
+olcArgsFile: ${LDAP_BASE_DIR}/var/run/slapd.args
+olcPidFile: ${LDAP_BASE_DIR}/var/run/slapd.pid
+
+#
+# Load dynamic backend modules commonly compiled in and available by default:
+#
+dn: cn=module,cn=config
+objectClass: olcModuleList
+cn: module
+olcModulePath: ${LDAP_BASE_DIR}/lib/openldap
+olcModuleLoad: back_mdb.la
+#olcModuleload: back_ldap.la
+#olcModuleload: back_passwd.la
+
+#
+# Schema settings
+#
+
+dn: cn=schema,cn=config
+objectClass: olcSchemaConfig
+cn: schema
+
+include: file://${LDAP_BASE_DIR}/etc/schema/core.ldif
+
+#
+# Frontend settings
+#
+
+dn: olcDatabase=frontend,cn=config
+objectClass: olcDatabaseConfig
+objectClass: olcFrontendConfig
+olcDatabase: frontend
+
+#
+# Configuration database
+#
+
+dn: olcDatabase=config,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: config
+olcAccess: to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" manage by * none
+
+#
+# Server status monitoring
+#
+
+dn: olcDatabase=monitor,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: monitor
+olcAccess: to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" read by dn.base="cn=Manager,dc=my-domain,dc=com" read by * none
+
+#
+# Backend database definitions
+#
+dn: olcDatabase=mdb,cn=config
+objectClass: olcDatabaseConfig
+objectClass: olcMdbConfig
+olcDatabase: mdb
+olcDbMaxSize: 1073741824
+olcSuffix: dc=my-domain,dc=com
+olcRootDN: cn=Manager,dc=my-domain,dc=com
+olcMonitoring: FALSE
+olcDbDirectory: ${LDAP_DATA_DIR}
+olcDbIndex: objectClass eq,pres
+olcDbIndex: ou,cn,mail,surname,givenname eq,pres,sub
+
+EOF
+
+}
+
+########################
+# Create LDAP online configuration
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_create_online_configuration() {
+    info "Creating LDAP online configuration"
+
+    ldap_create_slapd_file
+    if ! am_i_root; then
+        replace_in_file "${LDAP_SHARE_DIR}/slapd.ldif" "uidNumber=0" "uidNumber=$(id -u)"
+        replace_in_file "${LDAP_SHARE_DIR}/slapd.ldif" "gidNumber=0" "gidNumber=$(id -g)"
+    fi
+    local -a flags=(-F "$LDAP_ONLINE_CONF_DIR" -n 0 -l "${LDAP_SHARE_DIR}/slapd.ldif")
+    if am_i_root; then
+        debug_execute run_as_user "$LDAP_DAEMON_USER" slapadd "${slapd_debug_args[@]}" "${flags[@]}"
+    else
+        debug_execute slapadd "${slapd_debug_args[@]}" "${flags[@]}"
+    fi
+}
+
+########################
+# Configure LDAP credentials for admin user
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_admin_credentials() {
+    info "Configure LDAP credentials for admin user"
+    cat > "${LDAP_SHARE_DIR}/admin.ldif" << EOF
+dn: olcDatabase={2}mdb,cn=config
```

Comment:
You might do all of them in the same op?

Reply:
I don't get the suggestion here.